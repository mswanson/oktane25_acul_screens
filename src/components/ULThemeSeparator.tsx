import { cn } from "@/lib/utils";

export interface ULThemeSeparatorProps {
  /**
   * Optional text to display in the middle of the separator
   */
  text?: string;
  /**
   * Optional class names for additional styling or overriding default styles
   */
  className?: string;
}

const ULThemeSeparator = ({ text, className }: ULThemeSeparatorProps) => {
  // Base styles
  const containerStyles = "relative flex items-center my-4";

  // Theme overrides for line
  const themedLineStyles = cn(
    "flex-grow border-t",
    "theme-universal:border-input-border"
  );

  // Theme overrides for text
  const themedTextStyles = cn(
    "flex-shrink px-2",
    "theme-universal:text-body-text",
    "theme-universal:text-(length:--ul-theme-font-body-text-size)",
    "theme-universal:font-body"
  );

  if (text) {
    return (
      <div className={cn(containerStyles, className)}>
        <div className={themedLineStyles} />
        <span className={themedTextStyles}>{text}</span>
        <div className={themedLineStyles} />
      </div>
    );
  }

  return <hr className={cn("my-4", themedLineStyles, className)} />;
};

export default ULThemeSeparator;
