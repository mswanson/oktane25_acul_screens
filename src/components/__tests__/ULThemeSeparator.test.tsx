import { render, screen } from "@testing-library/react";

import ULThemeSeparator from "@/components/ULThemeSeparator";

describe("ULThemeSeparator Component", () => {
  // Snapshot Tests
  it("matches snapshot without text (renders as hr)", () => {
    const { container } = render(<ULThemeSeparator />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("matches snapshot with text (renders with separator text)", () => {
    const { container } = render(<ULThemeSeparator text="OR" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("matches snapshot with custom className", () => {
    const { container } = render(
      <ULThemeSeparator text="OR" className="custom-class" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  // Rendering Tests
  it("renders as hr element when no text is provided", () => {
    const { container } = render(<ULThemeSeparator />);
    const hrElement = container.querySelector("hr");
    expect(hrElement).toBeInTheDocument();
  });

  it("renders with separator text when text is provided", () => {
    render(<ULThemeSeparator text="OR" />);
    expect(screen.getByText("OR")).toBeInTheDocument();
  });

  it("renders separator lines when text is provided", () => {
    const { container } = render(<ULThemeSeparator text="OR" />);
    const lines = container.querySelectorAll("div.border-t");
    expect(lines).toHaveLength(2); // Two separator lines
  });

  // Theme Classes Tests
  it("applies theme classes correctly", () => {
    render(<ULThemeSeparator text="OR" />);
    const textElement = screen.getByText("OR");

    // Check for theme classes
    expect(textElement).toHaveClass("theme-universal:text-body-text");
    expect(textElement).toHaveClass("theme-universal:font-body");
  });
});
