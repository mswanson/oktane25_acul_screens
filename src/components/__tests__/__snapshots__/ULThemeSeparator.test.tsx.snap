// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ULThemeSeparator Component matches snapshot with custom className 1`] = `
<div
  class="relative flex items-center my-4 custom-class"
>
  <div
    class="flex-grow border-t theme-universal:border-input-border"
  />
  <span
    class="flex-shrink px-2 theme-universal:text-body-text theme-universal:text-(length:--ul-theme-font-body-text-size) theme-universal:font-body"
  >
    OR
  </span>
  <div
    class="flex-grow border-t theme-universal:border-input-border"
  />
</div>
`;

exports[`ULThemeSeparator Component matches snapshot with text (renders with separator text) 1`] = `
<div
  class="relative flex items-center my-4"
>
  <div
    class="flex-grow border-t theme-universal:border-input-border"
  />
  <span
    class="flex-shrink px-2 theme-universal:text-body-text theme-universal:text-(length:--ul-theme-font-body-text-size) theme-universal:font-body"
  >
    OR
  </span>
  <div
    class="flex-grow border-t theme-universal:border-input-border"
  />
</div>
`;

exports[`ULThemeSeparator Component matches snapshot without text (renders as hr) 1`] = `
<hr
  class="my-4 flex-grow border-t theme-universal:border-input-border"
/>
`;
